### PR TITLE
RELATED: RAIL-3679 add ExportWidget command

### PR DESCRIPTION
**Pull request by no23reason**
Adds a command to export an individual widget to CSV or XLSX.
This uses a new internal state slice that stores the executionResults.

---

Supported PR commands:

| Command                  | Description            |
| ------------------------ | ---------------------- |
| `ok to test`             | Re-run standard checks |
| `extended test`          | BackstopJS tests       |
| `extended check sonar`   | SonarQube tests        |
| `extended check cypress` | Cypress E2E tests      |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5089/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6779/ : SUCCESS in 19m 55s

File: libs/sdk-ui-dashboard/src/model/commandHandlers/widgets/exportInsightWidgetHandler.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+): SagaIterator<void> {
+    const { config, ref } = cmd.payload;
+
+    let canExport = false;
+    if (config.format === "csv") {
+        canExport = yield select(selectWidgetIsExportableToCsvByWidgetRef(ref));
+    }
+
+    if (config.format === "xlsx") {
+        canExport = yield select(selectWidgetIsExportableToXlsxByWidgetRef(ref));
```
Comment (ivanmjartan) [inline libs/sdk-ui-dashboard/src/model/commandHandlers/widgets/exportInsightWidgetHandler.ts:46]: This is weird  that we have permission together with result. Could we have two selector for it ?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5099/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6790/ : SUCCESS in 16m 27s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5103/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6794/ : SUCCESS in 14m 31s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5112/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6802/ : SUCCESS in 16m 34s

File: libs/sdk-ui-dashboard/src/model/commandHandlers/widgets/exportInsightWidgetHandler.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+export function* exportInsightWidgetHandler(
+    ctx: DashboardContext,
+    cmd: ExportInsightWidget,
+): SagaIterator<DashboardInsightWidgetExportResolved> {
+    const { config, ref } = cmd.payload;
+
+    yield put(insightWidgetExportRequested(ctx, ref, config, cmd.correlationId));
+
+    const isExportable: ReturnType<ReturnType<typeof selectIsExecutionResultReadyForExportByRef>> =
```
Comment (ivanmjartan) [inline libs/sdk-ui-dashboard/src/model/commandHandlers/widgets/exportInsightWidgetHandler.ts:68]: Could you separate isExportable into method and name it like validateExecResult ... 

Than just call validateExecResult same way as validateSettingsAndPermissions

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/5114/ : SUCCESS in 15s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6803/ : SUCCESS in 22m 31s

**Review by ivanmjartan — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1715

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1454/ : SUCCESS in 1m 14s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1731/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/517/ : SUCCESS in 18m 49s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6804/ : SUCCESS in 23m 10s

**Pull request merged**